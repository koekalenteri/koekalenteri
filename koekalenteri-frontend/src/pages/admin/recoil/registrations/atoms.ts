import { Registration, RegistrationGroup } from "koekalenteri-shared/model"
import { atom, atomFamily } from "recoil"

import { getRegistrations } from "../../../../api/registration"
import { logEffect, storageEffect } from "../../../recoil"


export interface RegistrationWithMutators extends Registration {
  setGroup: (group?: RegistrationGroup) => void
}

export const adminRegistrationIdAtom = atom<string | undefined>({
  key: 'adminRegistrationId',
  default: undefined,
  effects: [
    logEffect,
    storageEffect,
  ],
})

export const eventRegistrationsAtom = atomFamily<Registration[], string>({
  key: 'eventRegistrations',
  default: (eventId) => getRegistrations(eventId),
  effects: [
    logEffect,
    storageEffect,
  ],
})
