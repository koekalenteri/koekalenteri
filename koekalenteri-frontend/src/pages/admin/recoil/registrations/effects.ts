import type { Registration } from 'koekalenteri-shared/model'
import type { AtomEffect } from 'recoil'

import { getRegistrations } from '../../../../api/registration'
import { getParamFromFamilyKey, idTokenAtom } from '../../../recoil'

const loaded: Record<string, boolean> = {}

export const remoteRegistrationsEffect: AtomEffect<Registration[]> = ({ getPromise, node, setSelf, trigger }) => {
  if (trigger === 'get') {
    const eventId = getParamFromFamilyKey(node.key)
    if (loaded[eventId]) return
    loaded[eventId] = true
    getPromise(idTokenAtom).then((token) => {
      getRegistrations(eventId, token)
        .then((registrations) => {
          if (registrations) {
            setSelf(registrations)
          }
        })
        .catch((reason) => {
          console.error(reason)
          setSelf([])
        })
    })
  }
}
