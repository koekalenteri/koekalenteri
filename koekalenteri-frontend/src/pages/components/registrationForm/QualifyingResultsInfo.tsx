import { useCallback, useMemo } from 'react'
import { useTranslation } from 'react-i18next'
import { AddOutlined } from '@mui/icons-material'
import { Button, Grid } from '@mui/material'
import { ManualTestResult, QualifyingResult, Registration, TestResult } from 'koekalenteri-shared/model'

import { EventResultRequirementsByDate } from '../../../rules'
import CollapsibleSection from '../CollapsibleSection'

import QualifyingResultRow from './qualifyingResultsInfo/QualifyingResultRow'
import { createMissingResult, getResultId } from './qualifyingResultsInfo/utils'

type QualifyingResultsInfoProps = {
  regNo?: string
  results?: ManualTestResult[]
  requirements?: EventResultRequirementsByDate
  qualifyingResults?: QualifyingResult[]
  error?: boolean
  helperText?: string
  onChange?: (props: Partial<Registration>) => void
  onOpenChange?: (value: boolean) => void
  open?: boolean
}

export default function QualifyingResultsInfo({
  regNo,
  results,
  requirements,
  qualifyingResults,
  error,
  helperText,
  onChange,
  onOpenChange,
  open,
}: QualifyingResultsInfoProps) {
  const { t } = useTranslation()
  const disableResultInput = !requirements?.rules.length || !regNo

  const qualifying = useMemo(() => {
    if (!regNo) {
      return []
    }
    const newResults: Array<ManualTestResult> = (qualifyingResults ?? []).map((r) => ({
      ...r,
      id: getResultId(r),
      regNo,
    }))
    if (results) {
      for (const result of results) {
        if (!newResults.find((r) => !r.official && r.id && r.id === result.id)) {
          newResults.push({ ...result, official: false })
        }
      }
    }
    return newResults
  }, [qualifyingResults, results, regNo])

  const handleChange = useCallback(
    (result: ManualTestResult, props: Partial<TestResult>) => {
      const index = qualifying.findIndex((r) => !r.official && r.id && r.id === result.id)
      if (index >= 0) {
        const newResults: ManualTestResult[] = qualifying.slice(0)
        newResults.splice(index, 1, { ...result, ...props })
        onChange?.({ results: newResults.filter((r) => !r.official) })
      }
    },
    [onChange, qualifying]
  )

  const handleAddResult = useCallback(
    () =>
      onChange?.({
        results: [...(results ?? []), createMissingResult(requirements, qualifying, regNo ?? '')],
      }),
    [onChange, results, requirements, qualifying, regNo]
  )

  const handleRemoveResult = useCallback(
    (result: ManualTestResult) => {
      onChange?.({
        results: (results ?? []).filter((r) => r.id !== result.id),
      })
    },
    [onChange, results]
  )

  return (
    <CollapsibleSection
      title={t('registration.qualifyingResults')}
      error={error}
      helperText={helperText}
      open={open && !!regNo}
      onOpenChange={onOpenChange}
    >
      <Grid item container spacing={1}>
        {qualifying.map((result, index) => (
          <QualifyingResultRow
            key={getResultId(result)}
            result={result}
            requirements={requirements}
            onChange={handleChange}
            onRemove={handleRemoveResult}
          />
        ))}
        <Button startIcon={<AddOutlined />} disabled={disableResultInput} onClick={handleAddResult}>
          {t('registration.cta.addResult')}
        </Button>
      </Grid>
    </CollapsibleSection>
  )
}
