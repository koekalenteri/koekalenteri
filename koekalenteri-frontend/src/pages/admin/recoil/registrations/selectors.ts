import { Registration, RegistrationGroup } from 'koekalenteri-shared/model'
import { selector } from 'recoil'

import { adminEventIdAtom, eventClassAtom } from '../events/atoms'

import { adminRegistrationIdAtom, eventRegistrationsAtom } from './atoms'

export interface RegistrationWithMutators extends Registration {
  setGroup: (group?: RegistrationGroup) => void
}

export const currentEventRegistrationsSelector = selector<Registration[]>({
  key: 'currentEventRegistrations',
  get: ({ get }) => {
    const currentEventId = get(adminEventIdAtom)
    return currentEventId ? get(eventRegistrationsAtom(currentEventId)) : []
  },
  set: ({ get, set }, newValue) => {
    const currentEventId = get(adminEventIdAtom)
    if (currentEventId) {
      set(eventRegistrationsAtom(currentEventId), newValue)
    }
  },
})

export const currentEventClassRegistrationsSelector = selector<RegistrationWithMutators[]>({
  key: 'currentEventClassRegistrations',
  get: ({ get, getCallback }) => {
    const eventClass = get(eventClassAtom)
    const registrations = get(currentEventRegistrationsSelector)
    return registrations.filter(r => r.class === eventClass || r.eventType === eventClass).map(r => ({
      ...r,
      setGroup: getCallback(({ set }) => async (group?: RegistrationGroup) => {
        const newList = [...registrations]
        const index = newList.findIndex(item => item.id === r.id)
        if (index !== -1) {
          newList.splice(index, 1, { ...r, group })
        }
        set(currentEventRegistrationsSelector, newList)
      }),
    }))
  },
})

export const currentAdminRegistrationSelector = selector<Registration|undefined>({
  key: 'currentAdminRegistration',
  get: ({get}) => {
    const registrationId = get(adminRegistrationIdAtom)
    return get(currentEventClassRegistrationsSelector).find(r => r.id === registrationId)
  },
})
