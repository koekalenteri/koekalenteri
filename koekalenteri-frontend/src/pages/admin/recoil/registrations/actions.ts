import type { Registration, RegistrationGroupInfo } from 'koekalenteri-shared/model'

import { useSnackbar } from 'notistack'
import { useRecoilState, useRecoilValue } from 'recoil'

import { putRegistration, putRegistrationGroups } from '../../../../api/registration'
import { idTokenAtom } from '../../../recoil'
import { adminEventSelector } from '../events'

import { currentEventRegistrationsSelector } from './selectors'

export const useAdminRegistrationActions = (eventId: string) => {
  const [adminRegistrations, setAdminRegistrations] = useRecoilState(currentEventRegistrationsSelector)
  const [event, setEvent] = useRecoilState(adminEventSelector(eventId))
  const token = useRecoilValue(idTokenAtom)
  const { enqueueSnackbar } = useSnackbar()

  const updateAdminRegistration = (saved: Registration) => {
    const regs = [...adminRegistrations]
    const index = regs.findIndex((r) => r.id === saved.id)
    const insert = index === -1
    regs.splice(insert ? regs.length : index, insert ? 0 : 1, saved)
    setAdminRegistrations([...regs])
  }

  return {
    async save(reg: Registration) {
      const saved = await putRegistration(reg.ownerHandles ? { ...reg, handler: { ...reg.owner } } : reg, token)
      updateAdminRegistration(saved)
      return saved
    },

    async saveGroups(eventId: string, groups: RegistrationGroupInfo[]) {
      try {
        const { items, classes, entries, invitedOk, invitedFailed, pickedOk, pickedFailed, reserveOk, reserveFailed } =
          await putRegistrationGroups(eventId, groups, token)

        if (pickedOk.length) {
          enqueueSnackbar('Koepaikkailmoitus l√§hetetty onnistuneesti\n\n' + pickedOk.join('\n'), {
            variant: 'success',
            style: { whiteSpace: 'pre-line' },
          })
        }
        if (invitedOk.length) {
          enqueueSnackbar('Koekutsu l√§hetetty onnistuneesti\n\n' + invitedOk.join('\n'), {
            variant: 'success',
            style: { whiteSpace: 'pre-line' },
          })
        }
        if (reserveOk.length) {
          enqueueSnackbar('Varasijailmoitus l√§hetetty onnistuneesti\n\n' + reserveOk.join('\n'), {
            variant: 'success',
            style: { whiteSpace: 'pre-line' },
          })
        }
        if (pickedFailed.length) {
          enqueueSnackbar('Koepaikkailmoituksen l√§hetys ep√§onnistui üí©\n\n' + pickedFailed.join('\n'), {
            variant: 'success',
            style: { whiteSpace: 'pre-line' },
          })
        }
        if (invitedFailed.length) {
          enqueueSnackbar('Koekutsun l√§hetys ep√§onnistui üí©\n\n' + invitedFailed.join('\n'), {
            variant: 'success',
            style: { whiteSpace: 'pre-line' },
          })
        }
        if (pickedFailed.length) {
          enqueueSnackbar('Varasijailmoituksen l√§hetys ep√§onnistui üí©\n\n' + reserveFailed.join('\n'), {
            variant: 'success',
            style: { whiteSpace: 'pre-line' },
          })
        }
        setAdminRegistrations(items)
        if (event) {
          setEvent({ ...event, classes, entries })
        }
      } catch (e) {
        console.error(e)
        return false
      }
    },
  }
}
