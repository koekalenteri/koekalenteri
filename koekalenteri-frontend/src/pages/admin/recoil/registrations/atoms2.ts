import type { Registration } from 'koekalenteri-shared/model'

import { atomFamily } from 'recoil'

import { logEffect, storageEffect } from '../../../recoil'

import { adminEventRegistrationSelector } from './selectors'

export const editableAdminEventRegistrationByEventIdAndIdAtom = atomFamily<
  Registration | undefined,
  { eventId: string; id: string }
>({
  key: 'editableAdminEventRegistration/eventId+Id',
  default: adminEventRegistrationSelector,
  effects: [logEffect, storageEffect],
})
