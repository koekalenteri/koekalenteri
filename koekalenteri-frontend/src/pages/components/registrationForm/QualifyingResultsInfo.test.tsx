import { ReactNode } from 'react'
import { LocalizationProvider } from '@mui/x-date-pickers'
import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns'
import { render, screen } from '@testing-library/react'
import { fi } from 'date-fns/locale'
import { Registration } from 'koekalenteri-shared/model'

import { registrationWithStaticDates, registrationWithStaticDatesAndClass } from '../../../__mockData__/registrations'

import QualifyingResultsInfo from './QualifyingResultsInfo'

const Provider = ({ children }: { children: ReactNode }) => <LocalizationProvider dateAdapter={AdapterDateFns} adapterLocale={fi}>{children}</LocalizationProvider>

describe('QualifyingResultsInfo', () => {
  it('should render with minimal input', () => {
    const { container } = render(<QualifyingResultsInfo reg={{} as Registration} />, { wrapper: Provider })

    expect(container).toMatchSnapshot()
  })

  it('should render a NOME-B ALO registraton', () => {
    const { container } = render(<QualifyingResultsInfo reg={registrationWithStaticDatesAndClass} />, { wrapper: Provider })

    expect(registrationWithStaticDatesAndClass.eventType).toEqual('NOME-B')
    expect(registrationWithStaticDatesAndClass.class).toEqual('ALO')
    expect(container).toMatchSnapshot()
  })

  it('should not allow entering results for NOU test', async () => {
    render(<QualifyingResultsInfo reg={registrationWithStaticDates} />, { wrapper: Provider })

    const button = screen.getByRole('button', { name: 'registration.cta.addResult' })
    expect(button).toBeDisabled()
  })

  it('should allow entering results for NOME-B rest', async () => {
    render(<QualifyingResultsInfo reg={registrationWithStaticDatesAndClass} />, { wrapper: Provider })

    const button = screen.getByRole('button', { name: 'registration.cta.addResult' })
    expect(button).toBeEnabled()
  })
})
