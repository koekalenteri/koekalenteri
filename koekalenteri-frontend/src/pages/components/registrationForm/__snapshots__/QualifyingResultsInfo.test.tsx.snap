// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QualifyingResultsInfo should render a NOME-B ALO registraton 1`] = `
<div>
  <div
    class="MuiBox-root css-z0stvt"
  >
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeSmall css-2rshpx-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="KeyboardArrowDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <div
      class="MuiBox-root css-1qbrvqz"
    >
      <div
        class="MuiBox-root css-yxbw19"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          registration.qualifyingResults
        </p>
        <p
          class="MuiFormHelperText-root css-gorocw-MuiFormHelperText-root"
        />
      </div>
      <div
        class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
          >
            <div
              class="MuiBox-root css-n2y79c"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-1 css-v8g4am-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-1 css-1bzbk8b-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <div
                      class="MuiAutocomplete-root MuiAutocomplete-fullWidth MuiAutocomplete-hasPopupIcon css-r7d7so-MuiAutocomplete-root"
                      data-testid="testResult.eventType"
                    >
                      <div
                        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                      >
                        <label
                          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-disabled MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
                          data-shrink="true"
                          for=":r0:"
                          id=":r0:-label"
                        >
                          testResult.eventType
                        </label>
                        <div
                          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
                        >
                          <input
                            aria-autocomplete="list"
                            aria-expanded="false"
                            aria-invalid="false"
                            autocapitalize="none"
                            autocomplete="off"
                            class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                            disabled=""
                            id=":r0:"
                            role="combobox"
                            spellcheck="false"
                            type="text"
                            value="NOU"
                          />
                          <div
                            class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
                          >
                            <button
                              aria-label="Open"
                              class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                              disabled=""
                              tabindex="-1"
                              title="Open"
                              type="button"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                                data-testid="ArrowDropDownIcon"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M7 10l5 5 5-5z"
                                />
                              </svg>
                            </button>
                          </div>
                          <fieldset
                            aria-hidden="true"
                            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                          >
                            <legend
                              class="css-14lo706"
                            >
                              <span>
                                testResult.eventType
                              </span>
                            </legend>
                          </fieldset>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <div
                      class="MuiAutocomplete-root MuiAutocomplete-fullWidth MuiAutocomplete-hasPopupIcon css-1wijyiz-MuiAutocomplete-root"
                      data-testid="testResult.result"
                    >
                      <div
                        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                      >
                        <label
                          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-disabled MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
                          data-shrink="true"
                          for=":r2:"
                          id=":r2:-label"
                        >
                          testResult.result
                        </label>
                        <div
                          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
                        >
                          <input
                            aria-autocomplete="list"
                            aria-expanded="false"
                            aria-invalid="false"
                            autocapitalize="none"
                            autocomplete="off"
                            class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                            disabled=""
                            id=":r2:"
                            role="combobox"
                            spellcheck="false"
                            type="text"
                            value="NOU1"
                          />
                          <div
                            class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
                          >
                            <button
                              aria-label="Open"
                              class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                              disabled=""
                              tabindex="-1"
                              title="Open"
                              type="button"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                                data-testid="ArrowDropDownIcon"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M7 10l5 5 5-5z"
                                />
                              </svg>
                            </button>
                          </div>
                          <fieldset
                            aria-hidden="true"
                            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                          >
                            <legend
                              class="css-14lo706"
                            >
                              <span>
                                testResult.result
                              </span>
                            </legend>
                          </fieldset>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <div
                      class="MuiFormControl-root css-1q3rvyc-MuiFormControl-root"
                    >
                      <div
                        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
                      >
                        <label
                          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-disabled MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
                          data-shrink="true"
                          for=":r4:"
                          id=":r4:-label"
                        >
                          testResult.date
                        </label>
                        <div
                          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
                        >
                          <input
                            aria-invalid="false"
                            class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                            disabled=""
                            id=":r4:"
                            placeholder="dd.mm.yyyy"
                            type="tel"
                            value="10.01.2021"
                          />
                          <div
                            class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                          >
                            <button
                              aria-label="Choose date, selected date is 10. tammi 2021"
                              class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                              disabled=""
                              tabindex="-1"
                              type="button"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                                data-testid="CalendarIcon"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                                />
                              </svg>
                            </button>
                          </div>
                          <fieldset
                            aria-hidden="true"
                            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                          >
                            <legend
                              class="css-14lo706"
                            >
                              <span>
                                testResult.date
                              </span>
                            </legend>
                          </fieldset>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <div
                      class="MuiFormControl-root MuiTextField-root css-1r4zqtc-MuiFormControl-root-MuiTextField-root"
                    >
                      <label
                        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-disabled MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
                        data-shrink="true"
                        for=":r5:"
                        id=":r5:-label"
                      >
                        testResult.location
                      </label>
                      <div
                        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                          disabled=""
                          id=":r5:"
                          type="text"
                          value="test location"
                        />
                        <fieldset
                          aria-hidden="true"
                          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                        >
                          <legend
                            class="css-14lo706"
                          >
                            <span>
                              testResult.location
                            </span>
                          </legend>
                        </fieldset>
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <div
                      class="MuiFormControl-root MuiTextField-root css-8n9l5v-MuiFormControl-root-MuiTextField-root"
                    >
                      <label
                        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-disabled MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
                        data-shrink="true"
                        for=":r6:"
                        id=":r6:-label"
                      >
                        testResult.judge
                      </label>
                      <div
                        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                          disabled=""
                          id=":r6:"
                          type="text"
                          value="test judge"
                        />
                        <fieldset
                          aria-hidden="true"
                          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                        >
                          <legend
                            class="css-14lo706"
                          >
                            <span>
                              testResult.judge
                            </span>
                          </legend>
                        </fieldset>
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item css-1uydgkw-MuiGrid-root"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="DeleteOutlineIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9zm7.5-5-1-1h-5l-1 1H5v2h14V4z"
                          />
                        </svg>
                      </span>
                      registration.cta.deleteResult
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="AddOutlinedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                      />
                    </svg>
                  </span>
                  registration.cta.addResult
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`QualifyingResultsInfo should render with minimal input 1`] = `
<div>
  <div
    class="MuiBox-root css-z0stvt"
  >
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeSmall css-2rshpx-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="KeyboardArrowDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <div
      class="MuiBox-root css-1qbrvqz"
    >
      <div
        class="MuiBox-root css-yxbw19"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          registration.qualifyingResults
        </p>
        <p
          class="MuiFormHelperText-root css-gorocw-MuiFormHelperText-root"
        />
      </div>
      <div
        class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
          >
            <div
              class="MuiBox-root css-n2y79c"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-1 css-v8g4am-MuiGrid-root"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <span
                    class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="AddOutlinedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                      />
                    </svg>
                  </span>
                  registration.cta.addResult
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
